### PR TITLE
feat: simplify deployment steps and instructions

### PR DESCRIPTION
**Pull request by stvstnfrd**
Assuming the `README` was previously up-to-date and that I've sufficiently grokked it, this should offer a simplified interface for deploying with Heroku.

A stage deploy (from master) should be as simple as:

```sh
make deploy-stage
```

and prod:

```sh
make deploy-prod
```

You can also deploy from the current branch:

```sh
make deploy-stage-branch
```

or from an arbitrary branch:

```sh
make deploy-stage-branch DEPLOY_STAGING_COMMIT=feat/my-branch
```

Configuration/setup will be automatically run, if needed.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -94,3 +94,50 @@ rq-requeue-failed: ## Requeue failed RQ jobs
 
 rqinfo: ## See RQ info
 	@$(MAKE) cmd=rqinfo rq-cmd
+
+DEPLOY_STAGING_BRANCH_NAME=heroku
```
Comment (stvstnfrd) [inline Makefile:98]: This should actually be `*_REMOTE_NAME`.

File: Makefile [hunk 2]
```diff
@@ -94,3 +94,50 @@ rq-requeue-failed: ## Requeue failed RQ jobs
...
+DEPLOY_STAGING_APP_NAME=openedx-webhooks-staging
+DEPLOY_PROD_APP_NAME=openedx-webhooks
+DEPLOY_STAGING_COMMIT=HEAD
```
Comment (stvstnfrd) [inline Makefile:101]: Nit: Should this be `DEPLOY_STAGING_COMMIT`?

**Review by stvstnfrd — COMMENTED** [review]
TL;DR: Simplify workflow by minimizing the number of commands to be run, ie, a single `Makefile` command instead of several `heroku` commands.

File: Makefile [hunk 3]
```diff
@@ -94,3 +94,50 @@ rq-requeue-failed: ## Requeue failed RQ jobs
...
+	heroku releases --app "$(DEPLOY_STAGING_APP_NAME)" -n 1 2>/dev/null
+	heroku releases --app "$(DEPLOY_PROD_APP_NAME)" -n 1 2>/dev/null
+	@echo
+
+deploy-stage:  ## deploy master to stage via heroku
+	make deploy-stage-branch DEPLOY_STAGING_COMMIT=master
+
+deploy-stage-branch: deploy-check  ## deploy a branch to stage via heroku
+	@echo
+	# git push "$(DEPLOY_STAGING_BRANCH_NAME)" "$(DEPLOY_STAGING_COMMIT):master"
```
Comment (stvstnfrd) [inline Makefile:131]: TODO: Uncomment prior to merge.

File: Makefile [hunk 4]
```diff
@@ -94,3 +94,50 @@ rq-requeue-failed: ## Requeue failed RQ jobs
...
+	@echo
+	heroku releases --app "$(DEPLOY_STAGING_APP_NAME)" -n 1 2>/dev/null
+	heroku open --app "$(DEPLOY_STAGING_APP_NAME)" 2>/dev/null
+
+deploy-prod: deploy-check  ## deploy master to production via heroku
+	@echo
+	# heroku pipelines:promote -r "$(DEPLOY_STAGING_BRANCH_NAME)"
```
Comment (stvstnfrd) [inline Makefile:138]: TODO: Uncomment prior to merge.

**Review by stvstnfrd — COMMENTED** [review]
Remember to uncomment lines.

File: Makefile (hunk 2 above)
Comment (nedbat) [reply Makefile:101]: It is `DEPLOY_STAGING_COMMIT`.  You mean `DEPLOY_STAGING_BRANCH` maybe?

**Comment by nedbat** [comment]
This looks good.  Are you ready to merge it?

File: Makefile (hunk 1 above)
Comment (stvstnfrd) [reply Makefile]: Done.

File: Makefile (hunk 2 above)
Comment (stvstnfrd) [reply Makefile]: Ugh, yup! Done.

File: Makefile (hunk 3 above)
Comment (stvstnfrd) [reply Makefile]: Done.

File: Makefile (hunk 4 above)
Comment (stvstnfrd) [reply Makefile]: Done.

**Review by stvstnfrd — COMMENTED** [review]
Feedback addressed, comments removed;
this is ready to merge and use @nedbat
